user-interface/src/main/java/life/qbic/datamanager/views/projects/purchase/UploadPurchaseDialog.java

```diff
@@ -0,0 +1,193 @@
+    // Title box configuration
+    titleBox = new Div();
+    var title = new Span("Upload your offer");
+    title.addClassName("title");
+    titleBox.add(title);
+    titleBox.addClassName("title-box");
+
+    // Upload restriction display configuration
+    var restrictions = new Div();
+    restrictions.addClassName("restrictions");
+    restrictions.add(new Span("Accepted file formats: PDF (.pdf)"));
+    restrictions.add(
+        new Span("Maximum file size: %s MB".formatted(MAX_FILE_SIZE_BYTES / (1024 * 1024))));
+
+    // Uploaded purchase items display configuration
+    uploadedPurchaseItems = new Div();
+    uploadedPurchaseItems.addClassName("uploaded-documents");
+    disclaimer = new Div();
+    disclaimer.addClassName("instructions");
+    disclaimer.add(new Span("Set offer signed status"));
+    disclaimer.add(new Paragraph("Please tick the checkbox if the offer is signed."));
+
+
+    // Add upload offers to the purchase item section, where users can set the signed flag
+    upload.addSucceededListener(succeededEvent -> {
+      var purchase = new PurchaseItem(succeededEvent.getFileName());
+      uploadedPurchaseItems.add(purchase);
+      purchaseItemsCache.add(purchase);
+      toggleFileSectionIfEmpty();
+    });
+
+    // Synchronise the Vaadin upload component with the purchase list display
+    // When a file is removed  from the upload component, we also want to remove it properly from memory
+    // and from any additional display
+    upload.getElement().addEventListener("file-remove", this::processClientFileRemoveEvent)
+        .addEventData(VAADIN_FILENAME_EVENT);
+
+    // Put the elements together
+    add(titleBox, upload, restrictions, disclaimer, uploadedPurchaseItems);
+    addClassName("purchase-item-upload");
+    confirmButton.setText("Save");
+
+    // Init the visibility rendering once
+    toggleFileSectionIfEmpty();
+  }
+
+  @Override
+  protected void onConfirmClicked(ClickEvent<Button> clickEvent) {
+    fireConfirmEvent();
+  }
+
+  @Override
+  protected void onCancelClicked(ClickEvent<Button> clickEvent) {
+    this.close();
+  }
+
+  private void fireConfirmEvent() {
+    var confirmEvent = new ConfirmEvent<>(this, true);
+    confirmListeners.forEach(listener -> listener.onComponentEvent(confirmEvent));
+  }
+
+  private void processClientFileRemoveEvent(DomEvent event) {
+    JsonObject jsonObject = event.getEventData();
+    var fileName = jsonObject.getString(VAADIN_FILENAME_EVENT);
+    removeFile(fileName);
+  }
+
+  private void removeFile(String fileName) {
+    removeFileFromBuffer(fileName);
+    removeFileFromDisplay(fileName);
+    toggleFileSectionIfEmpty();
+  }
+
+  private void toggleFileSectionIfEmpty() {
+    if (uploadedPurchaseItems.getChildren().toList().isEmpty()) {
+      uploadedPurchaseItems.addClassName("hidden");
+      disclaimer.addClassName("hidden");
+    } else {
+      uploadedPurchaseItems.removeClassName("hidden");
+      disclaimer.removeClassName("hidden");
```

Comment:
you can use vaadins `setVisible(true)`.
This way you can get rid of the if-else as well.